akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/internal/TestInboxImpl.scala

```diff
@@ -53,3 +54,13 @@ private[akka] final class TestInboxImpl[T](path: ActorPath)
   @InternalApi private[akka] def as[U]: TestInboxImpl[U] = this.asInstanceOf[TestInboxImpl[U]]
 
 }
+
+@InternalApi
+object TestInboxImpl {
+  def apply[T](name: String): TestInboxImpl[T] = {
+    val uid = ThreadLocalRandom.current().nextInt()
```

Comment:
This is unlikely to lead to test failures, but could a counter (maybe guarded by synchronization (which would probably(?) be elided by JIT in the single-thread case)?) work here?

Reply:
Why is the instance id needed now, could we just leave it undefined instead? (We don't set it when creating a TestInbox in the behavior testkit)

Reply:
Just retaining the current behavior.  Perhaps it's because there's no preventing multiple inboxes from having the same name, though if nothing else cares about that?

Reply:
Yeah, I don't think that is important, since there is no global registry/tree where these lives.